modules/reactor/src/meshgenerators/FlexiblePatternGenerator.C

```diff
@@ -553,9 +563,24 @@ FlexiblePatternGenerator::FlexiblePatternGenerator(const InputParameters & param
 std::unique_ptr<MeshBase>
 FlexiblePatternGenerator::generate()
 {
-  if (isParamValid("external_boundary_id"))
-    MooseMesh::changeBoundaryId(
-        **_build_mesh, OUTER_SIDESET_ID, getParam<boundary_id_type>("external_boundary_id"), false);
+  if (_external_boundary_id != OUTER_SIDESET_ID)
+    MooseMesh::changeBoundaryId(**_build_mesh, OUTER_SIDESET_ID, _external_boundary_id, false);
+  if (!_external_boundary_name.empty())
+  {
+    // Check if _external_boundary_name has been assigned to another boundary id
+    const auto external_id_by_name =
+        (*_build_mesh)->get_boundary_info().get_id_by_name(_external_boundary_name);
+    if ((external_id_by_name != Moose::INVALID_BOUNDARY_ID) &&
+        (external_id_by_name != _external_boundary_id))
+      mooseError("External boundary name " + _external_boundary_name + " is associated with id " +
```

Comment:
Use paramError on external_boundary_name?

Reply:
Also "already associated" then output that it differs from the user-specified external_boundary_id

Reply:
Updated error message